### DISCUSSION TITLE
server error with google_drive ocr

### DISCUSSION THREAD

**Discussion by elsenchai** [xulihang/ImageTrans-docs#233] [Q&A] [answered]
title 
api key slot is using this link: http://service.basiccat.org:8090/ocr
google ocr is working with my own api key

what am i missing here? thanks

**Comment by xulihang** [comment]
If you have your own key, you can just use google instead of google drive.

Google drive has advantages likes being free. But it is a bit complicated to configure. If you do not publish the app, the token generated is only valid for 7 days.

The default url of google drive uses my own google drive to upload images and then OCR. The token has expired so it says server error.

If you need to use google drive with your own credentials, you can check out this issue: https://github.com/xulihang/ImageTrans-docs/issues/121

**Reply by elsenchai** [reply]
thanks for the reply, i'll read up the links you provided
i use google drive as a second option ocr as it could grab the text when google would sometimes fail to do so

**Comment by elsenchai** [comment]
i tried using the cli method but im stucked at the second step where it shows:

Authorization Error
Error 400: redirect_uri_mismatch
You can't sign in to this app because it doesn't comply with Google's OAuth 2.0 policy.
If you're the app developer, register the redirect URI in the Google Cloud Console.step, 

Ive so many questions, I've tried running the command but nothing happens on my PC. Where do I upload the image and how to change the image's mimetype? What URI should i put for the OAuth 2.0 Client IDs ?

**Reply by xulihang** [reply]
Create a credential for desktop:

![image](https://user-images.githubusercontent.com/5462205/177681993-a0422da0-bcaa-4210-b1af-ed9a4a808b73.png)

**Reply by xulihang** [reply]
You don't need to set the mimetype. It is done by the cli tool. I just write there for insights.

**Comment by elsenchai** [comment, marked as answer]
Things i made for it to work, given you have normal google ocr running:
Enable Google Drive API
Publish the app
Put credentials into main folder after the cloud app is running + googleocrplugin.jar and googleocrplugin.xml in plugins folder
